db/db.go

```diff
@@ -1295,14 +1107,14 @@ func saveValidators(data *types.EpochData, tx *sqlx.Tx, client rpc.Client) error
 					WHEN EXCLUDED.exitepoch <= %[1]d THEN 'exited'
 					WHEN EXCLUDED.activationeligibilityepoch = 9223372036854775807 THEN 'deposited'
 					WHEN EXCLUDED.activationepoch > %[1]d THEN 'pending'
-					WHEN EXCLUDED.slashed AND EXCLUDED.activationepoch < %[1]d AND GREATEST(COALESCE(validators.lastattestationslot, 0), EXCLUDED.lastattestationslot) < %[2]d THEN 'slashing_offline'
+					WHEN EXCLUDED.slashed AND EXCLUDED.activationepoch < %[1]d AND GREATEST(COALESCE(validators.lastattestationslot / %[4]d, 0), EXCLUDED.lastattestationslot / %[4]d) < %[2]d THEN 'slashing_offline'
```

Comment:
Why do you need the GREATEST func here?

Reply:
We get here when an existing `validatorindex` has to be updated. The whole `ON CONFLICT` block takes the greater value for `lastattestationslot` from either the existing or the new data. This is even done when `lastattestationslot` is set within `ON CONFLICT` in line 1103.

I interpret this as "use the latest of both available `lastattestationslots` as it will be the truly latest one". But to be honest, this handling was here before and I did not question it too much.